### PR TITLE
[libremidi] Add new port

### LINKED ISSUE #20623: [New Port Request] <libremidi>
Library name: libremidi

Library description: A modern C++ MIDI real-time & file I/O library. Supports Windows, macOS, Linux and WebMIDI.

Source repository URL: https://github.com/jcelerier/libremidi

Anything else that is useful to know when adding (such as optional features the library may have that should be included): this is a fork of RtMidi, and since RtMidi is already in vcpkg, I imagine the starting point to port would be just copy and paste the config files from https://github.com/microsoft/vcpkg/tree/master/ports/rtmidi and tweak them.

### PR DESCRIPTION
**Pull request by marinedalek**
Fixes #20623

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

Opened as draft for testing/comment

### CODE REVIEW

File: ports/libremidi/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,17 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO jcelerier/libremidi
+    REF "v4.0"
```
Comment (jimwang118) [inline ports/libremidi/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"

```

File: ports/libremidi/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/libremidi PACKAGE_NAME libremidi)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(INSTALL "${SOURCE_PATH}/LICENSE.md" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (jimwang118) [inline ports/libremidi/portfile.cmake:16]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.md")
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Review by jimwang118 — APPROVED** [review]

**Comment by jimwang118** [comment]
Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

**Comment by jcelerier** [comment]
yay ^-^

**Comment by BillyONeal** [comment]
We accidentally merged this without verifying:

> Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all find_package calls are REQUIRED, are satisfied by vcpkg.json's declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)

and the port breaks that rule:

```
## JACK support ##
if(NOT LIBREMIDI_NO_JACK)
  find_path(WEAKJACK_PATH weakjack/weak_libjack.h HINTS ${WEAKJACK_FOLDER})
  find_path(JACK_PATH jack/jack.h)
  if(WEAKJACK_PATH AND JACK_PATH)
    message(STATUS "libremidi: using WeakJACK")
    set(LIBREMIDI_HAS_JACK 1)

    target_include_directories(libremidi ${_public} $<BUILD_INTERFACE:${WEAKJACK_PATH}> $<BUILD_INTERFACE:${JACK_PATH}>)
  elseif(JACK_PATH)
    find_library(JACK_LIBRARIES jack)
    if(JACK_LIBRARIES)
      message(STATUS "libremidi: using linked JACK")
      set(LIBREMIDI_HAS_JACK 1)
      
      target_link_libraries(libremidi ${_public} ${JACK_LIBRARIES})
      target_include_directories(libremidi ${_public} $<BUILD_INTERFACE:${JACK_PATH}>)
    endif()
  endif()

  if(LIBREMIDI_HAS_JACK)
    target_compile_definitions(libremidi ${_public} LIBREMIDI_JACK)
  endif()
endif()
```

and as a result this port fails in CI because jack2 happened to be installed first.

I submitted https://github.com/microsoft/vcpkg/pull/34223 which should fix this.

**Comment by splisp** [comment]
I have commented on another issue but maybe this one is more relevant. The link to my comment is:

https://github.com/microsoft/vcpkg/issues/20623#issuecomment-1774173080

**Pull request merged**